html/pfappserver/lib/pfappserver/Form/Config/Scan.pm

```diff
@@ -0,0 +1,236 @@
+   label => 'Password',
+   required => 1,
+   password => 0,
+   messages => { required => 'You must specify the password' },
+  );
+
+
+has_field 'type' =>
+  (
+   type => 'Select',
+   label => 'Scan Type',
+   options_method => \&options_type,
+  );
+
+has_field 'categories' =>
+  (
+   type => 'Select',
+   multiple => 1,
+   label => 'Roles',
+   options_method => \&options_categories,
+   element_class => ['chzn-select'],
+   element_attr => {'data-placeholder' => 'Click to add a role'},
+   tags => { after_element => \&help,
+             help => 'Nodes with the selected roles will be affected' },
+  );
+
+has_field 'dot1x' =>
+  (
+   type => 'Checkbox',
+   label => '802.1x',
+   tags => { after_element => \&help,
+             help => 'If this option is enabled, PacketFence will scan all the 802.1x auto-registration connections.' },
+  );
+
+has_field 'dot1x_type' =>
+  (
+   type => 'Text',
+   label => '802.1x types',
+   tags => { after_element => \&help,
+             help => 'Comma-delimited list of EAP-Type attributes that will pass to the scan engine.' },
+  );
+
+has_field 'duration' =>
+  (
+   type => 'Duration',
+   label => 'Duration',
+   default => '20s',
+   tags => { after_element => \&help,
+             help => 'Approximate duration of a scan. User being scanned on registration are presented a progress bar for this duration, afterwards the browser refreshes until scan is complete.' },
+  );
+
+has_field 'registration' =>
+  (
+   type => 'Checkbox',
+   label => 'Scan on registration',
+   tags => { after_element => \&help,
+             help => 'If this option is enabled, the PF system will scan each host after registration is complete.' },
+  );
+
+has_field 'pre_registration' =>
+  (
+   type => 'Checkbox',
+   label => 'Scan before registration',
+   tags => { after_element => \&help,
+             help => 'If this option is enabled, the PF system will scan host before the registration.' },
+  );
+
+has_field 'post_registration' =>
+  (
+   type => 'Checkbox',
+   label => 'Scan after registration',
+   tags => { after_element => \&help,
+             help => 'If this option is enabled, the PF system will scan host after on the production vlan.' },
+  );
+
+has_field 'oses' =>
+  (
+   type => 'Select',
+   multiple => 1,
+   label => 'OS',
+   options_method => \&options_oses,
+   element_class => ['chzn-deselect'],
+   element_attr => {'data-placeholder' => 'Click to add an OS'},
+   tags => { after_element => \&help,
+             help => 'Nodes with the selected OS will be affected' },
+  );
+
+=head2 options_oses
+
+=cut
+
+sub options_oses {
+    my $self = shift;
+    return $self->form->oses;
+}
+
+=head2 options_type
+
+Dynamically extract the descriptions from the various Scan modules.
+
+=cut
+
+sub options_type {
+    my $self = shift;
+
+    my %paths = ();
+    my $wanted = sub {
+        if ((my ($module, $pack, $scan) = $_ =~ m/$lib_dir\/((pf\/scan\/([A-Z0-9][\w\/]+))\.pm)\z/)) {
```

Comment:
Ok let's leave it as it then, we should eventually move this to a pf::util function though